### ISSUE TITLE
Troubles running manage mode

### ISSUE THREAD

**Issue by aanm** [docker/swarm#939]
I'm running swarm agent and swarm manager on 2 computers, they can see each other and then can reach the internet. However if I use the manager mode it doesn't work, but it does work if I use the static mode. Is there any problem on running swarm agent and swarm manager with tokens at the same time?
```bash
node1 ~$ sudo docker logs swarm-manager-token
time="2015-06-12T17:03:56Z" level=info msg="Listening for HTTP" addr=":2375" proto=tcp 
time="2015-06-12T17:11:01Z" level=info msg="Listening for HTTP" addr=":2375" proto=tcp 
node1 ~$ sudo docker logs swarm-manager-static-ip
time="2015-06-12T17:38:23Z" level=info msg="Listening for HTTP" addr=":2375" proto=tcp 
time="2015-06-12T17:38:23Z" level=info msg="Registered Engine node2 at 192.168.50.6:2375" 
time="2015-06-12T17:38:23Z" level=info msg="Registered Engine node1 at 192.168.50.5:2375" 
node1 ~$ sudo  docker run --rm swarm list token://<token>
192.168.50.5:2375
192.168.50.6:2375
```

**Comment by aluzzardi** [comment]
Can you reach those nodes from the machine running the manager?

```
docker -H 192.168.50.6:2375 info
docker -H 192.168.50.5:2375 info
```

**Comment by aanm** [comment]
@aluzzardi Yes. The `swarm-manager-static` has the following startup command `manage 192.168.50.5:2375, 192.168.50.6:2375` and I can successfully run those 2 commands that you ask me. Is there any chance of running a debug mode or something?

**Comment by aluzzardi** [comment]
@aanm Yes, swarm outputs log messages to stdout. If you are running swarm in a container, just do `docker logs <swarm manage container id>`.

You can get more logs by setting the log level to debug: `swarm -l debug manage <...>`

**Comment by aluzzardi** [comment]
Sorry, I misread the entire issue :) Let me read it again

**Comment by aluzzardi** [comment]
In the meantime, please use `-l debug` in `swarm-manager-token` to see what happens

**Comment by aluzzardi** [comment]
Can you provide the start options of `swarm manage` and `swarm join`?

**Comment by aanm** [comment]
@aluzzardi So many questions let me see if I can get them 

node1 `swarm join`
```bash
"Cmd": [
     "join",
     "--addr=192.168.50.5:2375",
     "token://<token>"
],
```
node1 `swarm manage`
```bash
  -p 2370:2375 \
  -d \
  swarm:0.3.0-rc2 \
  -l debug manage token://<token>
```
node2 `swarm join`
```bash
"Cmd": [
     "join",
     "--addr=192.168.50.6:2375",
     "token://<token>"
],
```
node2 `swarm manage`
```bash
  -p 2370:2375 \
  -d \
  swarm:0.3.0-rc2 \
  -l debug manage token://<token>
```

node1 manage logs
```bash
time="2015-06-12T23:50:43Z" level=debug msg="Initializing discovery service" name=token uri="token" 
time="2015-06-12T23:50:43Z" level=debug msg="Initializing strategy" name=spread 
time="2015-06-12T23:50:43Z" level=debug msg="Initializing filter" name=affinity 
time="2015-06-12T23:50:43Z" level=debug msg="Initializing filter" name=health 
time="2015-06-12T23:50:43Z" level=debug msg="Initializing filter" name=constraint 
time="2015-06-12T23:50:43Z" level=debug msg="Initializing filter" name=port 
time="2015-06-12T23:50:43Z" level=debug msg="Initializing filter" name=dependency 
time="2015-06-12T23:50:43Z" level=debug msg="Initializing cluster" name=swarm 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/json" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/events" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/images/search" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/images/get" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/images/{name:.*}/get" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/images/{name:.*}/history" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/containers/json" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/export" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/info" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/images/viz" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/top" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/logs" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/stats" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/_ping" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/images/json" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/images/{name:.*}/json" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/attach/ws" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/exec/{execid:.*}/json" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/version" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/containers/ps" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/changes" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/images/create" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/images/load" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/images/{name:.*}/push" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/images/{name:.*}/tag" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/pause" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/rename" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/exec/{execid:.*}/start" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/commit" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/exec/{execid:.*}/resize" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/restart" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/unpause" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/kill" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/start" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/stop" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/resize" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/create" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/build" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/wait" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/attach" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/copy" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/exec" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=POST route="/auth" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=DELETE route="/containers/{name:.*}" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=DELETE route="/images/{name:.*}" 
time="2015-06-12T23:50:43Z" level=debug msg="Registering HTTP route" method=OPTIONS route= 
time="2015-06-12T23:50:43Z" level=info msg="Listening for HTTP" addr=":2375" proto=tcp
```
node2 manage logs
```bash
time="2015-06-12T23:50:44Z" level=debug msg="Initializing discovery service" name=token uri="token" 
time="2015-06-12T23:50:44Z" level=debug msg="Initializing strategy" name=spread 
time="2015-06-12T23:50:44Z" level=debug msg="Initializing filter" name=affinity 
time="2015-06-12T23:50:44Z" level=debug msg="Initializing filter" name=health 
time="2015-06-12T23:50:44Z" level=debug msg="Initializing filter" name=constraint 
time="2015-06-12T23:50:44Z" level=debug msg="Initializing filter" name=port 
time="2015-06-12T23:50:44Z" level=debug msg="Initializing filter" name=dependency 
time="2015-06-12T23:50:44Z" level=debug msg="Initializing cluster" name=swarm 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=DELETE route="/containers/{name:.*}" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=DELETE route="/images/{name:.*}" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=OPTIONS route= 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/changes" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/logs" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/attach/ws" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/images/search" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/containers/json" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/export" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/json" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/top" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/containers/{name:.*}/stats" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/info" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/version" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/images/viz" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/containers/ps" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/images/{name:.*}/get" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/images/{name:.*}/history" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/images/{name:.*}/json" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/exec/{execid:.*}/json" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/_ping" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/events" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/images/json" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=GET route="/images/get" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/start" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/stop" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/resize" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/create" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/kill" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/wait" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/attach" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/copy" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/exec" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/auth" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/build" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/images/load" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/images/{name:.*}/push" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/images/{name:.*}/tag" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/pause" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/rename" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/exec/{execid:.*}/start" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/commit" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/images/create" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/exec/{execid:.*}/resize" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/unpause" 
time="2015-06-12T23:50:44Z" level=debug msg="Registering HTTP route" method=POST route="/containers/{name:.*}/restart" 
time="2015-06-12T23:50:44Z" level=info msg="Listening for HTTP" addr=":2375" proto=tcp 
```

**Comment by aluzzardi** [comment]
Wait - you have two swarm manage running?

**Comment by aanm** [comment]
node1 IP -> 192.168.50.5
node2 IP -> 192.168.50.6
manager with discovery hub -> nodeIP:2370
manager with static IPs -> nodeIP:2373
```bash
vagrant@node2:~$ sudo docker -H 192.168.50.5:2370 info  
Containers: 0
Images: 0
Storage Driver: 
Strategy: spread
Filters: affinity, health, constraint, port, dependency
Nodes: 0
Execution Driver: 
Kernel Version: 
Operating System: 
CPUs: 0
Total Memory: 0 B
Name: 
ID: 
Http Proxy: 
Https Proxy: 
No Proxy: 

vagrant@node2:~$ sudo docker -H 192.168.50.5:2373 info 
Containers: 13
Images: 12
Storage Driver: 
Strategy: spread
Filters: affinity, health, constraint, port, dependency
Nodes: 2
 node1: 192.168.50.5:2375
  └ Containers: 7
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 1.019 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.13.0-24-generic, operatingsystem=Ubuntu 14.04 LTS, storagedriver=aufs
 node2: 192.168.50.6:2375
  └ Containers: 6
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 1.019 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.13.0-24-generic, operatingsystem=Ubuntu 14.04 LTS, storagedriver=aufs
Execution Driver: 
Kernel Version: 
Operating System: 
CPUs: 4
Total Memory: 2.038 GiB
Name: 
ID: 
Http Proxy: 
Https Proxy: 
No Proxy:

vagrant@node2:~$ sudo docker -H 192.168.50.6:2370 info  
Containers: 0
Images: 0
Storage Driver: 
Strategy: spread
Filters: affinity, health, constraint, port, dependency
Nodes: 0
Execution Driver: 
Kernel Version: 
Operating System: 
CPUs: 0
Total Memory: 0 B
Name: 
ID: 
Http Proxy: 
Https Proxy: 
No Proxy: 

vagrant@node2:~$ sudo docker -H 192.168.50.6:2373 info 
Containers: 13
Images: 12
Storage Driver: 
Strategy: spread
Filters: affinity, health, constraint, port, dependency
Nodes: 2
 node1: 192.168.50.5:2375
  └ Containers: 7
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 1.019 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.13.0-24-generic, operatingsystem=Ubuntu 14.04 LTS, storagedriver=aufs
 node2: 192.168.50.6:2375
  └ Containers: 6
  └ Reserved CPUs: 0 / 2
  └ Reserved Memory: 0 B / 1.019 GiB
  └ Labels: executiondriver=native-0.2, kernelversion=3.13.0-24-generic, operatingsystem=Ubuntu 14.04 LTS, storagedriver=aufs
Execution Driver: 
Kernel Version: 
Operating System: 
CPUs: 4
Total Memory: 2.038 GiB
Name: 
ID: 
Http Proxy: 
Https Proxy: 
No Proxy:
```

**Comment by vieux** [comment]
@aanm I get that you want to have 2 managers to test, one token and one IP. but it seems like you have 4, 2 per node ???

**Comment by aanm** [comment]
@vieux yes, yes I have. 1 manager and 1 agent on each node.

**Comment by vieux** [comment]
@aanm 

```
node1 IP -> 192.168.50.5
node2 IP -> 192.168.50.6
manager with discovery hub -> nodeIP:2370
manager with static IPs -> nodeIP:2373
```

so what's the IPs of `manager with discovery hub` and `manager with static IPs` ?

**Comment by aanm** [comment]
`manager with discovery hub`: 192.168.50.5:2370,192.168.50.6:2370
`manager with static IPs`: 192.168.50.5:2373,192.168.50.6:2373

**Comment by vieux** [comment]
wat

**Comment by vieux** [comment]
so you have 4 managers ?

**Comment by aanm** [comment]
lol why is that such a big fuss? Does it break anything?

**Comment by vieux** [comment]
@aanm no but it doesn't make sense

**Comment by vieux** [comment]
and it's hard to follow all the logs and everything. can you try to reduce the number or managers and see if it happen again ?

**Comment by aanm** [comment]
@vieux I know it doesn't... But that was the only way I found out to get docker events for the whole swarm on every node (by only doing docker -H localhost:2370 docker events). If you have any other idea I'm free to suggestions :-)

**Comment by aanm** [comment]
@vieux It does happen again. My main goal is what I've wrote in the last comment. I only added the managers with static IPs to see if the same thing happen.

**Comment by aanm** [comment]
Somehow I can't replicate the issue with a new setup, sorry
